exchanges/binance/binance_wrapper.go

```diff
@@ -534,8 +534,36 @@ func (b *Binance) GetRecentTrades(p currency.Pair, assetType asset.Item) ([]trad
 }
 
 // GetHistoricTrades returns historic trade data within the timeframe provided
-func (b *Binance) GetHistoricTrades(_ currency.Pair, _ asset.Item, _, _ time.Time) ([]trade.Data, error) {
-	return nil, common.ErrFunctionNotSupported
+func (b *Binance) GetHistoricTrades(p currency.Pair, a asset.Item, from, to time.Time) ([]trade.Data, error) {
+	req := AggregatedTradeRequestParams{
+		Symbol:    p,
```

Comment:
This is exactly what I'm talking about with the type safety - as the caller of the `GetAggregatedTrades` I don't want to care about this.
In my opinion the "clean" solution would be separation of the core concept - currency pair - and it's representation... again, think of the `time.Time` analogy. It represents the concept of an instance of time, but how it is represented in different contexts (on the wire, configuration) is different. This approach would work for the currency pairs as well. The separator character is only relevant in the context of specific serialisation format, but not as part of the concept of a currency pair.

Reply:
The clean solution would be to move the formatting/serialisation of pairs and built-in types to the API funcs instead of the wrappers. At the moment, the wrappers act as the "unified API" for all exchanges and add value like you say. They primarily serialise the supplied currency pair ("BTC-USDT", "btc_usdt", "BTC~~~YOLO420~~~USDT"; whether from gctcli or an external app) to the correct format the exchange requires ("BTCUSDT"). This is what engine uses to perform actions for all exchanges and why it's done in them. However, I do agree that we can instead move the type safety out of the wrappers and into the actual exchange API funcs, but this is out of the scope for this PR and will require a large overhaul. Am happy to leave your change but I can see that you've added a commit to keep it consistent with everything else.

Reply:
OK, let's keep it now...